src/core/simplify.rkt

```diff
@@ -20,19 +20,12 @@
 ;; the last expression is the simplest unless something went wrong due to unsoundness
 ;; if the input specifies proofs, it instead returns proofs for these expressions
 (define/contract (simplify-batch runner extractor)
-  (-> egg-runner? procedure? (listof (listof expr?)))
-  (timeline-push! 'inputs
-                  (map ~a (batch-extract-exprs (egg-runner-batch runner) (egg-runner-roots runner))))
+  (-> egg-runner? procedure? (listof batchref?))
+  (timeline-push! 'inputs (map ~a (batch->progs (egg-runner-batch runner) (egg-runner-roots runner))))
   (timeline-push! 'method "egg-herbie")
-
-  (define simplifieds (run-egg runner (cons 'single extractor)))
-  (define out
-    (for/list ([simplified simplifieds]
-               [expr (batch-extract-exprs (egg-runner-batch runner) (egg-runner-roots runner))])
-      (remove-duplicates (cons expr simplified))))
-
```

Comment:
As I understand, the only reason for this code with `expr` and `cons` is to avoid duplications in alt structure. This part is used only at one place in Herbie. Otherwise, the `simplified` option is taken at once.
Maybe it is alright to simplify the code?

Reply:
Well, I think it's a bit more complex than that but now that I think about it if the results aren't any worse then let's simplify the code.

Reply:
Hmm, but maybe the results _are_ worse? So the reason for the `(cons expr simplified)` is that if we simplify an expression into something less accurate, we want the original unsimplified expression to also be there. The `remove-duplicates`, yes, is unnecessary since we're already avoiding duplicates, but returning two things does make some sense.

Reply:
I agree, just changed it, I also think that it should be preserved as it is

Reply:
Indeed, that was one of the reasons of the output accuracy differences. 
The accuracy got better.
The last nightly run [main](https://nightly.cs.washington.edu/reports/herbie/1727778729:nightly:main:cd05a70693/), [artem-batch-finish-rewrite](https://nightly.cs.washington.edu/reports/herbie/1727806158:nightly:artem-batch-finish-rewrite:4e449e8630/)